src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2049,7 +2049,7 @@ public function getRouteKey()
      */
     public function getRouteKeyName()
     {
-        return $this->getKeyName();
+        return array_get(config('database'), 'route_key_name', $this->getKeyName());
```

Comment:
This also couples to the foundation component, so breaks all usage outside of laravel.